### DISCUSSION TITLE
Getting error "Couldn't inject pageWorld.js. Check that the extension is using MV3...." when trying to upgrade chrome extension from MV2 to MV3

### DISCUSSION THREAD

**Discussion by kinsflow** [InboxSDK/InboxSDK#1134] [Q&A] [answered]
**Question(s)**

I am trying to upgrade my chrome extension from MV2 to MV3. And in my extension, I have an integration with inboxSdk what has always worked while it was MV2. but since i upgraded to mv3, I now get the error

Error handling response: Error: Couldn't inject pageWorld.js. Check that the extension is using MV3 and has the correct permissions and host_permissions in its manifest.
![image](https://github.com/InboxSDK/InboxSDK/assets/36258984/071a293c-e320-48cb-9268-27e3d1239ba6)
meanwhile i am currently using inboxsdk version "^2.1.26"
![image](https://github.com/InboxSDK/InboxSDK/assets/36258984/40591afc-5bdd-4b5a-b3e6-236303c7ce37)

from the error, it speaks on my host_permission and my permission thingy. 
below are the SS of what I have 
![image](https://github.com/InboxSDK/InboxSDK/assets/36258984/48f6bdd4-cf22-4d80-a357-5e7a3b22beac)
![image](https://github.com/InboxSDK/InboxSDK/assets/36258984/a6e04649-b21b-4253-948d-c1b12eee8962)

What could be wrong? does this have to do with a bug on inboxSdk as a library? Anyhelp would be appreciated.

Thanks.

**Comment by wegry** [comment]
Hey @kinsflow, my suspicion is that you might not have `background.js` set in `background.service_worker` of your manifest.json. Here's what that setting looks like in one of our example extensions: https://github.com/InboxSDK/InboxSDK/blob/85b290630a069af517c363cbc44b7777d102b3d4/examples/mole-example/manifest.json#L22

**Reply by wegry** [reply]
https://github.com/InboxSDK/hello-world/blob/main/webpack.common.js#L10 shows one way to generate background.js.

**Reply by kinsflow** [reply]
@wegry , I actually do have a background.js fully set. see below an attachement

<img width="277" alt="image" src="https://github.com/InboxSDK/InboxSDK/assets/36258984/e8dc80cf-3968-4aa1-a7c0-ab5d98f04b41">

but unlike what the [example](https://github.com/InboxSDK/hello-world/blob/main/webpack.common.js#L10) you shared depict, i have my own custom background.js and i am not using that of inboxSdk's - `@inboxsdk/core/background.js` see attached below

<img width="579" alt="image" src="https://github.com/InboxSDK/InboxSDK/assets/36258984/6fd9b211-54cf-4e56-a8dd-6610e6df119f">

**Reply by wegry** [reply, marked as answer]
Ah, I see. This may be related to https://github.com/InboxSDK/hello-world/issues/10. The author includes their solution there as well.

**Comment by tipycalFlow** [comment]
This error still comes on `"@inboxsdk/core": "^2.2.1"` using the default `background.js` on closing gmail tab and opening again
